modules/exploits/linux/http/ivanti_connect_secure_rce_cve_2024_37404.rb

```diff
@@ -0,0 +1,428 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  class IvantiError < StandardError; end
+  class IvantiNoAccessError < IvantiError; end
+  class IvantiNotFoundError < IvantiError; end
+  class IvantiUnexpectedResponseError < IvantiError; end
+  class IvantiUnknownError < IvantiError; end
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti Connect Secure Authenticated Remote Code Execution via OpenSSL CRLF Injection',
+        'Description' => %q{
+          This module exploits a CRLF injection vulnerability in Ivanti Connect
+          Secure to achieve remote code execution (CVE-2024-37404). Versions
+          prior to 22.7R2.1 and 22.7R2.2 are vulnerable. Note that Ivanti Policy
+          Secure versions prior to 22.7R1.1 are also vulnerable but this module
+          doesn't support this software.
+
+          Valid administrative credentials are required. A non-administrative
+          user is also required and can be created using the administrative
+          account, if needed.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Richard Warren', # Vulnerability discovery and PoC
+          'Christophe De La Fuente', # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2024-37404'],
+          ['URL', 'https://attackerkb.com/topics/FI5vcuGwyM/cve-2024-37404'],
+          ['URL', 'https://forums.ivanti.com/s/article/Security-Advisory-Ivanti-Connect-Secure-and-Policy-Secure-CVE-2024-37404'],
+          ['URL', 'https://blog.amberwolf.com/blog/2024/october/cve-2024-37404-ivanti-connect-secure-authenticated-rce-via-openssl-crlf-injection/']
+        ],
+        'DisclosureDate' => '2024-10-08',
+        'Platform' => 'linux',
+        'Arch' => ARCH_X86,
+        'Privileged' => true, # Administrative access is needed and code execution as root.
+        'Targets' => [
+          ['Automatic', {}]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
```

Comment:
Could we add a new side effect here? e.g.
```suggestion
          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS, ACCOUNT_LOGOUT]
```

Reply:
I added a new side effect in [f5ac33e](https://github.com/rapid7/metasploit-framework/pull/19595/commits/f5ac33e3a752703734ce6af70caaa24f5612ac65). Thanks!